core/reactor_common.c

```diff
@@ -1655,47 +1658,156 @@ void usage(int argc, char* argv[]) {
 int default_argc = 0;
 char** default_argv = NULL;
 
+typedef struct federation_metadata_t {
+    char* federation_id;
+    char* rti_host;
+    int rti_port;
+    char* rti_user;
+} federation_metadata_t;
+
+federation_metadata_t federation_metadata = {
+    .federation_id =  "Unidentified Federation",
+    .rti_host = NULL,
+    .rti_port = -1,
+    .rti_user = NULL
+};
+
+/**
+ * Checks if port is valid.
+ * @return true if valid, false otherwise.
+ */
+bool validate_port(char* port) {
+    // magic number 6 since port range is [0, 65535]
+    int port_len = strnlen(port, 6); 
+    if (port_len < 1 || port_len > 5) {
+        return false;
+    }
+
+    for (int i = 0; i < port_len; i++) {
+        if (!isdigit(port[i])) {
+            return false;
+        }
+    }
+    int port_number = atoi(port);
+    return port_number >= 0 && port_number <= 65535;
+}
+
+/**
+ * A helper struct for passing rti_addr information betweem parse_rti_addr and extract_rti_addr_info
+ */
+typedef struct rti_addr_info_t {
+    char rti_hostStr[256];
+    char rti_portStr[6];
+    char rti_userStr[256];
+    bool has_host;
+    bool has_port;
+    bool has_user;
+} rti_addr_info_t;
+
 /**
- * The ID of the federation that this reactor will join.
- * This should be overridden with a command-line -i option to ensure
- * that each federate only joins its assigned federation.
+ * A helper enum for the returned status code of parse_rti_addr.  
  */
-char* federation_id = "Unidentified Federation";
+typedef enum parse_rti_code_t {
+    SUCCESS,
+    INVALID_PORT,
+    INVALID_HOST,
+    INVALID_USER,
+    FAILED_TO_PARSE
+} parse_rti_code_t;
+
+/**
+ * Extract the host, port and user from rti_addr.  
+ */
+void extract_rti_addr_info(char* rti_addr, rti_addr_info_t* rti_addr_info) {
+    int n = 0;
+    n = sscanf(rti_addr, "%255s@%255s:%5s", rti_addr_info->rti_userStr, rti_addr_info->rti_hostStr, rti_addr_info->rti_portStr);
+    if (n == 3) {
+        rti_addr_info->has_host = rti_addr_info->has_port = rti_addr_info->has_user = true;
+    } else {
+        n = sscanf(rti_addr, "%255s:%5s", rti_addr_info->rti_hostStr, rti_addr_info->rti_portStr);
```

Comment:
I think the logic here can be improved a bit. First, `sscanf` can return `EOF` or `-1`, which are both error conditions that need to be dealt with. Second, it seems that a `user@ip` argument from the user can be problematic here (note that we do have a default port number). Would it be safe here to check if any of the fields (e.g., `rti_userStr`) have been set, and then try to re-parse the string accordingly?

Reply:
I tried to use sscanf and multiple passes but could not get it to work. Since we handle the rti's address using regex in compile time, I used regex to parse the rti's address instead. This would help standardize how we handle RTI's address at different places.